regulations/static/regulations/js/source/views/main/child-view.js

```diff
@@ -151,15 +151,19 @@ var ChildView = Backbone.View.extend({
             var url = this.url,
                 hashPosition;
 
+            console.log(options);
+
             // if a hash has been passed in
             if (options && typeof options.scrollToId !== 'undefined') {
                 url += '#' + options.scrollToId;
-            }
-            else {
+            } else {
                 hashPosition = (typeof Backbone.history.fragment === 'undefined') ? -1 : Backbone.history.fragment.indexOf('#');
                 //  Be sure not to lose any hash info
                 if (hashPosition !== -1) {
-                    url += Backbone.history.fragment.substr(hashPosition);
+                    url = url.slice(0, hashPosition) + '#' + options.id;
```

Comment:
I know this is going to get looked at again, but I think we could remove the ternary operators in L160 to make lines 156-168 to be something like this:

```
if (options && typeof options.scrollToId !== 'undefined') {
                url += '#' + options.scrollToId;
            } else {
                // Don't lose the hash info.
                if (typeof Backbone.history.fragment !== 'undefined') {
                    hashPosition =  Backbone.history.fragment.indexOf('#');
                    url = url.slice(0, hashPosition) + '#' + options.id;
                } else if (options.type !== 'diff')  {
                    url += '#' + options.id;
                }
            }
```